src/call.c

```diff
@@ -2165,6 +2165,87 @@ static bool valid_addressfamily(struct call *call, const struct stream *strm)
 }
 
 
+/**
+ * Find a call given the value of a Replaces header
+ *
+ * @param calls   List of calls
+ * @param hdr     Value of Replaces header
+ *
+ * @return Call object if found, NULL if not found
+ */
+static struct call *call_find_replaces(const struct list *calls,
+				       const struct pl *hdr)
+{
+	int err;
+	struct le *le;
+	struct sip_msg msg = { 0 };
+	char *hdr_val;
+	const char *p, *p2;
+	struct call *ret = NULL;
+
+	if (!calls || !pl_isset(hdr))
+		return NULL;
+
+	err = pl_strdup(&hdr_val, hdr);
+	if (err)
+		return NULL;
+
+	msg.req = true;
+	msg.callid.p = hdr->p;
+	msg.callid.l = strchr(hdr_val, ';')
+		? (size_t)(strchr(hdr_val, ';') - hdr->p)
+		: hdr->l;
+
+	if (!msg.callid.l)
+		goto out;
+
+	/* Find to-tag */
+	if ((p = strchr(hdr_val, ';'))) {
+		if ((p = str_str(p, "to-tag="))) {
+			p += 7;  /* Skip "to-tag=" */
+			p2 = strchr(p, ';');
+			msg.to.tag.p = p;
+			msg.to.tag.l = p2 ? (size_t)(p2 - p) : str_len(p);
+		}
+	}
+
+	/* Find from-tag */
+	if ((p = strchr(hdr_val, ';'))) {
+		if ((p = str_str(p, "from-tag="))) {
```

Comment:
can re_regex used for `to-tag` and `from-tag` value detection?

Reply:
Oh, right. That's much simpler. Done.